src/base1/test-dbus-common.js

```diff
@@ -108,6 +108,39 @@ function common_dbus_tests(channel_options, bus_name)
             });
     });
 
+    QUnit.asyncTest("out of range integers", function() {
+        assert.expect(22);
+
+        var dbus = cockpit.dbus(bus_name, channel_options);
+
+        function testNumber(type, value) {
+            QUnit.stop();
+            dbus.call("/otree/frobber", "org.freedesktop.DBus.Properties",
+                      "Set", [ "com.redhat.Cockpit.DBusTests.Frobber", type, { t: type, v: value } ]).
+                fail(function(ex) {
+                    assert.equal(ex.name, "org.freedesktop.DBus.Error.InvalidArgs");
+                }).
+                always(function() {
+                    assert.equal(this.state(), "rejected", "rejected out of bounds number");
+                    QUnit.start();
+                });
+        }
+
+        testNumber('y', -1);
+        testNumber('y', 0xff + 1);
+        testNumber('n', -0x8000 -1);
+        testNumber('n', 0x7fff + 1);
+        testNumber('q', -1);
+        testNumber('q', 0xffff + 1);
+        testNumber('i', -0x80000000 -1);
+        testNumber('i', 0x7fffffff + 1);
+        testNumber('u', -1);
+        testNumber('u', 0xffffffff + 1);
+        testNumber('t', -1);
+
+        QUnit.start();
```

Comment:
I've added a couple of positive test cases, with numbers that are bigger than the next smaller type. Large `guint64` values aren't supported, because json-glib only gives us `gint64`. This might be something we need to address in the future, but these are mostly used for microsecond time stamps, so we've got a bit of time ;)